src/languages/mathematica.js

```diff
@@ -6,61 +6,118 @@ Website: https://www.wolfram.com/mathematica/
+    begin: /\\\[/,
+    end: /[$a-zA-Z][$a-zA-Z0-9]+]/
   };
 
-  const VARIABLES = {
-    className: 'symbol',
-    begin: '(?:\\$|[a-zA-Z])[a-zA-Z0-9\\$]*',
-    relevance: 0
+  const OPERATORS = {
+    className: 'operator',
+    begin: /[+\-*/,;.:@~=><&|_`'^?!%]+/
+  };
+
+  const PATTERNS_AND_SLOTS = {
+    className: 'pattern',
+    variants: [
+      {
+        begin: /([a-zA-Z$][a-zA-Z0-9$]*)?_+([a-zA-Z$][a-zA-Z0-9$]*)?/,
+        keywords: {
+          $pattern: symbol_re,
+          strong: Mathematica.SYSTEM_SYMBOLS.join(" ")
```

Comment:
Please explain this variant and why it's keywords would be "strong".... typically we try to use semantic use of classes, not visual... so this seems strange or else I simply don't understand what is happening here.

Reply:
Yes, I'll post a screenshot. Basically, when we have a parameter of a function, Mathematica users love to see this in green. When this parameter has a type-specification like `Integer` which is a keyword, it should stay green but bold like keywords. This decision is not final and I'd like to discuss this with the community. It's quite possible that the `keywords` section in this goes away entirely.

Reply:
>  When this parameter has a type-specification like Integer which is a keyword, it should stay green but bold like keywords.

But that's why we have "keyword" and variants - if it's a special type of keyword then we need to find the proper variant, not overload `strong`.  `strong` simply means "strong/bold text" (as in something like Markdown or perhaps Latex)... also it's **not always rendered bold**. Themes can choose to render it however, and sometimes just use a color.  That's what I mean when I say that these styles are semantic, not visual.